new build fail looks real
```
ERROR: test_report_module (corehq.apps.app_manager.tests.test_suite.SuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/test_suite.py", line 784, in test_report_module
    app.create_suite(),
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/models.py", line 4035, in create_suite
    return suite_xml.SuiteGenerator(self).generate_suite()
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/suite_xml.py", line 711, in generate_suite
    map(add_to_suite, self.sections)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/suite_xml.py", line 709, in add_to_suite
    getattr(suite, attr).extend(getattr(self, attr))
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 127, in __call__
    value = self.func( *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/suite_xml.py", line 1155, in details
    for detail_type, detail, enabled in module.get_details():
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/models.py", line 2999, in get_details
    for details in config.get_details():
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/models.py", line 2819, in get_details
    yield (self.summary_detail_id, self.summary_details(), True)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/models.py", line 2904, in summary_details
    ] + list(_get_graph_fields())
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/models.py", line 2847, in _get_graph_fields
    for chart_config in self.report.charts:
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 127, in __call__
    value = self.func( *args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/models.py", line 278, in charts
    return [ChartFactory.from_spec(g._obj) for g in self.configured_charts]
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/reports/factory.py", line 146, in from_spec
    return cls.spec_map[spec['type']].wrap(spec)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/reports/specs.py", line 333, in wrap
    obj['chart_id'] = (obj.get('title') or '') + str(hash(frozenset(sorted(obj.items()))))
TypeError: unhashable type: 'list'
```

--- a/corehq/apps/app_manager/static/app_manager/js/report-module.js
+++ b/corehq/apps/app_manager/static/app_manager/js/report-module.js
@@ -73,8 +73,7 @@ var ReportModule = (function () {
 
             var chartsToConfigs = {};
             var currentChartsToConfigs = self.currentGraphConfigs();
-            for (var chart_id in currentChartsToConfigs) {
-                var graph_config = currentChartsToConfigs[chart_id];
+            _.each(currentChartsToConfigs, function(graph_config, chart_id) {
                 chartsToConfigs[chart_id] = {
                     series_configs: {}
                 };
@@ -83,7 +82,7 @@ var ReportModule = (function () {
                 }
                 chartsToConfigs[chart_id].graph_type = graph_config.graph_type();
                 chartsToConfigs[chart_id].config = configToDict(graph_config.config);
-            }
+            });
             return chartsToConfigs;
         };
 

--- a/corehq/apps/app_manager/views.py
+++ b/corehq/apps/app_manager/views.py
@@ -3,7 +3,6 @@ import copy
 import logging
 import hashlib
 import itertools
-import uuid
 from django.utils.decorators import method_decorator
 from djangular.views.mixins import allow_remote_invocation, JSONResponseMixin
 from lxml import etree

--- a/corehq/apps/userreports/reports/specs.py
+++ b/corehq/apps/userreports/reports/specs.py
@@ -1,3 +1,4 @@
+import json
 from django.utils.translation import ugettext as _
 from dimagi.ext.jsonobject import JsonObject, StringProperty, BooleanProperty, ListProperty, DictProperty, ObjectProperty
 from jsonobject.base import DefaultProperty
@@ -330,8 +331,7 @@ class ChartSpec(JsonObject):
     @classmethod
     def wrap(cls, obj):
         if obj.get('chart_id') is None:
-            # http://stackoverflow.com/questions/5884066/hashing-a-python-dictionary
-            obj['chart_id'] = (obj.get('title') or '') + str(hash(frozenset(sorted(obj.items()))))
+            obj['chart_id'] = (obj.get('title') or '') + str(hash(json.dumps(sorted(obj.items()))))
         return super(ChartSpec, cls).wrap(obj)
 
 

